vaadin-login-flow-parent/vaadin-login-flow/src/main/java/com/vaadin/flow/component/login/LoginOverlay.java

```diff
@@ -231,6 +237,130 @@ public String getDescription() {
+        private LoginOverlayCustomFields(LoginOverlay overlay) {
+            super("custom-fields", overlay);
+        }
+    }
+
+    /**
+     * Class for adding and removing components to the footer area of the
+     * overlay.
+     */
+    final public static class LoginOverlayFooter extends LoginOverlayContent {
+        private LoginOverlayFooter(LoginOverlay overlay) {
+            super("footer", overlay);
+        }
+    }
+
+    /**
+     * This class defines the common behavior for adding/removing components to
+     * the custom fields and footer parts.
+     */
+    abstract static class LoginOverlayContent implements Serializable {
+        private final LoginOverlay overlay;
+        private final String slot;
+
+        protected LoginOverlayContent(String slot, LoginOverlay overlay) {
+            this.slot = slot;
+            this.overlay = overlay;
+        }
+
+        /**
+         * Adds the given components to the container. Note: components have to
+         * be added when the overlay is closed.
+         *
+         * @param components
+         *            the components to be added.
+         */
+        public void add(Component... components) {
+            if (overlay.isOpened()) {
+                return;
+            }
+
+            Objects.requireNonNull(components, "Components should not be null");
+            for (Component component : components) {
+                Objects.requireNonNull(component,
+                        "Component to add cannot be null");
+                SlotUtils.addToSlot(overlay, slot, component);
+            }
+        }
+
+        /**
+         * Removes the given components from the container. Note: components
+         * have to be removed when the overlay is closed.
+         *
+         * @param components
+         *            the components to be removed.
+         */
+        public void remove(Component... components) {
+            if (overlay.isOpened()) {
+                return;
+            }
+
+            Objects.requireNonNull(components, "Components should not be null");
+            for (Component component : components) {
+                Objects.requireNonNull(component,
+                        "Component to remove cannot be null");
+                Element element = component.getElement();
+
+                if (overlay.getElement().equals(element.getParent())
+                        && Objects.equals(element.getAttribute("slot"), slot)) {
+                    element.removeAttribute("slot");
+                    overlay.getElement().removeChild(element);
+                }
+            }
+        }
+
+        /**
+         * Removes all components from the container. Note: components have to
+         * be removed when the overlay is closed.
+         */
+        public void removeAll() {
+            if (overlay.isOpened()) {
+                return;
+            }
+
+            SlotUtils.clearSlot(overlay, slot);
```

Comment:
Maybe out of the scope, but `SlotUtils.clearSlot` in `LoginOverlay.removeAll` does not remove the slot attributes, while `LoginOverlay.remove` removes them, and also `SlotUtils.addToSlot` in `LoginOverlay.add` adds them. Do you think it makes sense to update `SlotUtils.clearSlot` to also remove the slot attributes?

Reply:
Good point. I think we should align it. I'll create a separate PR to update `SlotUtils.clearSlot`.

Reply:
@ugur-vaadin Submitted #5429 for `SlotUtils` change that you suggested.